text/3772-build-target-edition.md

```diff
@@ -0,0 +1,180 @@
+
+...
+
+```toml
+[lib]
+name = "foo"           # The name of the target.
+path = "src/lib.rs"    # The source file of the target.
+test = true            # Is tested by default.
+doctest = true         # Documentation examples are tested by default.
+bench = true           # Is benchmarked by default.
+doc = true             # Is documented by default.
+proc-macro = false     # Set to `true` for a proc-macro library.
+harness = true         # Use libtest harness.
+crate-type = ["lib"]   # The crate types to generate.
+required-features = [] # Features required to build this target (N/A for lib).
+
+edition = "2015"       # Deprecated, N/A for Edition 20XX+
+```
+
+...
+
+### The `edition` field
+
+The `edition` field defines the [Rust edition] the target will use. If not
+specified, it defaults to the [`edition` field][package-edition] for the
+`[package]`.
+
+This field is deprecated and unsupported for Edition 20XX+
+
+## Migration guide
+
+*From [Rust Edition Guide: Advanced migration strategies](https://doc.rust-lang.org/nightly/edition-guide/editions/advanced-migrations.html#migrating-a-large-project-or-workspace)*
+
+### Migrating a large project or workspace
+
+You can migrate a large project incrementally to make the process easier if you run into problems.
+
+In a [Cargo workspace], each package defines its own edition, so the process naturally involves migrating one package at a time.
+
+Within a [Cargo package], you can either migrate the entire package at once, or migrate individual [Cargo targets] one at a time.
+For example, if you have multiple binaries, tests, and examples, you can use specific target selection flags with `cargo fix --edition` to migrate just that one target.
+By default, `cargo fix` uses `--all-targets`.
+
+*(removed talk of the build-target `edition` field)*
+
+### Migrating macros
+
+...
+
+If you have macros, you are encouraged to make sure you have tests that fully
+cover the macro's syntax. You may also want to test the macros by importing and
+using them in crates from multiple editions, just to ensure it works correctly
+everywhere.
+You can do this in doctests by setting the [edition attribute](https://doc.rust-lang.org/stable/rustdoc/write-documentation/documentation-tests.html#attributes)
+or by creating a package for your tests in your workspace for each edition.
+
+If you run into issues, you'll need to read through the chapters of
+this guide to understand how the code can be changed to work across all
+editions.
+
+*(added a testing strategy which was previously left unspoken)*
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+A non-`None` edition will be considered deprecated
+- A deprecation message will eventually be shown by Cargo
+  - Timing depends on if this will be blocked on having `[lints]` control over this or not
+- In Edition 20XX+, an error will be reported when this field is present
```

Comment:
@Lokathor  from https://github.com/rust-lang/rfcs/pull/3772#issuecomment-2657842887

> So the error case is if you set an edition >= X in your `package` and then also set any edition value at all in particular build targets?

Reply:
Yes. I will make that more clear.

Reply:
See 13702bb